src/main/java/de/tum/in/www1/artemis/service/ExerciseDeletionService.java

```diff
@@ -221,6 +221,9 @@ public void reset(Exercise exercise) {
      * @param exercise for which the plagiarism results and participations should be deleted
      */
     public void deletePlagiarismResultsAndParticipations(Exercise exercise) {
+        // delete all participant scores to avoid issues when deleting results later on
+        participantScoreRepository.deleteAllByExerciseIdTransactional(exercise.getId());
```

Comment:
Please verify with @krusche that the use of transactional is appropriate here.

Reply:
This was pushed by @krusche, and considering that this is a larger delete operation Transaction should be fine here.

Reply:
Adding to that: we did not add a new transactional here, but we simply use an existing one.

But most probably the nested transaction annotations led to problems in this particular case